stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetLoadingFragment.kt

```diff
@@ -17,20 +17,17 @@ class PaymentSheetLoadingFragment : Fragment(R.layout.fragment_payment_sheet_loa
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        if (activity == null) {
-            return
-        }
+        val activity = this.activity ?: return
 
-        activityViewModel.paymentMethods.observe(requireActivity()) { paymentMethods ->
+        activityViewModel.paymentMethods.observe(activity) { paymentMethods ->
             val target = if (paymentMethods.isEmpty()) {
                 TransitionTarget.AddPaymentMethodSheet
             } else {
                 TransitionTarget.SelectSavedPaymentMethod
             }
             activityViewModel.transitionTo(target)
         }
-        activityViewModel.updatePaymentMethods(requireActivity().intent)
-
-        // TODO: Fetch payment intent before transitioning
+        activityViewModel.updatePaymentMethods(activity.intent)
+        activityViewModel.fetchPaymentIntent(activity.intent)
```

Comment:
Should we wait until this completes before we call `activityViewModel.transitionTo` or is that just over-complicating things?

Not sure if there's an easy way to observe the combination of 2 LiveData's

Reply:
I think it's a question of when we want to show the loading indicator. I'll think about this.